### PR TITLE
Drop memory-fs to speed up reloads.

### PR DESCRIPTION
**Pull request by tido64**
Since we need to serialize payloads to communicate with worker processes over
WebSockets, we will not see any improvements using an in-memory file system as
serializing big debug bundles is a time-consuming process in JS.

Before:
image

After:
image

Addresses #493.

### CODE REVIEW

**Comment by tido64** [comment]
cc @thymikee, @zamotany

File: src/compiler/worker/initWorker.js [hunk 1]
```diff
@@ -82,7 +89,7 @@ module.exports = function initWorker({
       if (fs.existsSync(filename)) {
         send(Events.FILE_RECEIVED, {
           taskId: payload.taskId,
-          file: fs.readFileSync(filename),
+          file: filename,
```
Comment (thymikee) [inline src/compiler/worker/initWorker.js]: I think it makes sense now to rename the field to `filename` or `filePath`, pick whatever feels better :)

File: src/compiler/worker/runWebpackCompiler.js [hunk 2]
```diff
@@ -67,7 +67,6 @@ module.exports = function runWebpackCompiler({
     ],
   });
 
-  // Use memory fs
   compiler.outputFileSystem = fs;
```
Comment (thymikee) [inline src/compiler/worker/runWebpackCompiler.js]: Am I missing something or it seems like we could remove the `outputFileSystem` entirely? https://webpack.js.org/api/node/#custom-file-systems

Or at least we could simplify https://github.com/callstack/haul/pull/512/files#diff-99e77f1c262ae763f3d7d593b2c0b417R44 by spreading fs and adding `mkdirp` and `join` if that's necessary

Comment (tido64) [reply src/compiler/worker/runWebpackCompiler.js]: I added a comment here for future reference.

File: src/compiler/worker/runWebpackCompiler.js [hunk 3]
```diff
@@ -67,6 +67,8 @@ module.exports = function runWebpackCompiler({
     ],
   });
 
+  // As of Webpack 4.12.0, |outputFileSystem| must be set since there is no
+  // fallback despite the documentation stating otherwise.
```
Comment (thymikee) [inline src/compiler/worker/runWebpackCompiler.js]: Haha Webpack docs always being helpful on the topic 😄

Comment (tido64) [reply src/compiler/worker/runWebpackCompiler.js]: Yeah, my first attempt was removing this line. I was surprised when it simply crashed 😄 

Thanks for reviewing and merging it 🎉

**Pull request merged**